### PR TITLE
[CROSSDATA-351] [SECURITY] Session added to driver & common

### PR DESCRIPTION
**Pull request by miguel0afd**

### CODE REVIEW

File: driver/src/test/scala/com/stratio/crossdata/driver/FlattenedTablesIT.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ class FlattenedTablesIT extends MongoWithSharedContext {
   "The Driver" should " List table's description with nested and array fields flattened" in {
     assumeCrossdataUpAndRunning
 
-    val flattenedDriver = Driver(true)
+    val flattenedDriver = new Driver(flattenTables = true)
```
Comment (compae) [inline driver/src/test/scala/com/stratio/crossdata/driver/FlattenedTablesIT.scala]: more beautiful, all people understand this now!

**Comment by compae** [comment]
LGTM

**Comment by miguel0afd** [comment]
@darroyocazorla I've added your fantastic suggestions ;-)

File: driver/src/main/scala/com/stratio/crossdata/driver/session/SessionManager.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package com.stratio.crossdata.driver.session
+
+import java.util.UUID
+
+import com.stratio.crossdata.common.security.Session
+
+object SessionManager {
+
+  def createSession(auth: Authentication): Session ={
+    return Session(UUID.randomUUID.toString)
```
Comment (darroyocazorla) [inline driver/src/main/scala/com/stratio/crossdata/driver/session/SessionManager.scala]: image

def createSession(auth: Authentication): Session = Session(UUID.randomUUID.toString)

**Comment by darroyocazorla** [comment]
+1

**Comment by compae** [comment]
+1

File: driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala [hunk 3]
```diff
@@ -56,25 +47,52 @@ object Driver extends DriverConfig {
...
+  def this(serverHosts: java.util.List[String]) = this(
+    Map(DriverConfigHosts -> ConfigValueFactory.fromAnyRef(serverHosts)),
+    Driver.generateDefaultAuth,
+    false)
 
   import Driver._
 
+  val driverSession = SessionManager.createSession(auth)
+
+  def securitizeCommand(command: SQLCommand): SecureSQLCommand = {
```
Comment (pfcoperez) [inline driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala]: If you move this conversion to `Driver` companion object and change its signature to implicit `implicit def securitizeCommand(command: SQLCommand): SecureSQLCommand`, you can pass `SQLCommand` where `SecureSQLCommand` is required and the user won't even notice it

File: common/src/main/scala/com/stratio/crossdata/common/Messages.scala [hunk 4]
```diff
@@ -18,18 +18,18 @@ package com.stratio.crossdata.common
...
-                       queryId: UUID = UUID.randomUUID(),
-                       retrieveColumnNames: Boolean = false,
-                       timeout: Option[FiniteDuration] = None
-                     ) extends Command
+case class SQLCommand(query: String,
+                      queryId: UUID = UUID.randomUUID(),
+                      retrieveColumnNames: Boolean = false,
+                      timeout: Option[FiniteDuration] = None) extends Command
+
+case class SecureSQLCommand(sqlCommand: SQLCommand, session: Session)
```
Comment (pfcoperez) [inline common/src/main/scala/com/stratio/crossdata/common/Messages.scala]: I'd make a general envelope instead a concrete `SecureSQLCommand`. That is:
```
case class SecureCommand(cmd: Command, session: Session)
```

File: server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala [hunk 5]
```diff
@@ -35,9 +34,10 @@ class ServerActor(cluster: Cluster, xdContext: XDContext) extends Actor with Ser
   override lazy val logger = Logger.getLogger(classOf[ServerActor])
 
   def receive: Receive = {
-    case sqlCommand @ SQLCommand(query, _, withColnames, timeout) =>
-      logger.debug(s"Query received ${sqlCommand.queryId}: ${sqlCommand.query}. Actor ${self.path.toStringWithoutAddress}")
-      context.actorOf(JobActor.props(xdContext, sqlCommand, sender(), timeout))
+
+    case secureSQLCommand @ SecureSQLCommand(SQLCommand(query, queryId, withColnames, timeout), session) =>
```
Comment (pfcoperez) [inline server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala]: Why not?
```
case SecureSQLCommand(command @ SQLCommand(query, queryId, withColnames, timeout), session) =>
    context.actorOf(JobActor.props(xdContext, command, sender(), timeout))
```

**Comment by pfcoperez** [comment]
Beautiful! +1

File: common/src/main/scala/com/stratio/crossdata/common/Messages.scala (hunk 4 above)
Comment (miguel0afd) [inline common/src/main/scala/com/stratio/crossdata/common/Messages.scala]: Good idea!

File: server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala (hunk 5 above)
Comment (miguel0afd) [inline server/src/main/scala/com/stratio/crossdata/server/actors/ServerActor.scala]: Done!

File: driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala (hunk 3 above)
Comment (miguel0afd) [inline driver/src/main/scala/com/stratio/crossdata/driver/Driver.scala]: Sounds good! But, in this case, that would imply to move the auth to the companion object and that would be trickier.

**Pull request merged**